pkg/apiserver/apiserver.go

```diff
@@ -277,9 +274,13 @@ type StripVersionNegotiatedSerializer struct {
 	runtime.NegotiatedSerializer
 }
 
-func (n StripVersionNegotiatedSerializer) EncoderForVersion(serializer runtime.Serializer, gv unversioned.GroupVersion) runtime.Encoder {
-	encoder := n.NegotiatedSerializer.EncoderForVersion(serializer, gv)
-	return stripVersionEncoder{encoder, serializer}
+func (n StripVersionNegotiatedSerializer) EncoderForVersion(encoder runtime.Encoder, gv unversioned.GroupVersion) runtime.Encoder {
+	serializer, ok := encoder.(runtime.Serializer)
```

Comment:
if you need an interface that implements both, create that interface and use it.